ssv/types.go

```diff
@@ -33,37 +33,35 @@ type AttesterCalls interface {
 // ProposerCalls interface has all block proposer duty specific calls
 type ProposerCalls interface {
 	// GetBeaconBlock returns beacon block by the given slot and committee index
-	GetBeaconBlock(slot phase0.Slot, committeeIndex phase0.CommitteeIndex, graffiti, randao []byte) (*bellatrix.BeaconBlock, error)
+	GetBeaconBlock(slot phase0.Slot, graffiti, randao []byte) (*bellatrix.BeaconBlock, error)
 	// SubmitBeaconBlock submit the block to the node
 	SubmitBeaconBlock(block *bellatrix.SignedBeaconBlock) error
 }
 
 // AggregatorCalls interface has all attestation aggregator duty specific calls
 type AggregatorCalls interface {
 	// SubmitAggregateSelectionProof returns an AggregateAndProof object
-	SubmitAggregateSelectionProof(slot phase0.Slot, committeeIndex phase0.CommitteeIndex, slotSig []byte) (*phase0.AggregateAndProof, error)
+	SubmitAggregateSelectionProof(slot phase0.Slot, committeeIndex phase0.CommitteeIndex, committeeLength uint64, slotSig []byte) (*phase0.AggregateAndProof, error)
 	// SubmitSignedAggregateSelectionProof broadcasts a signed aggregator msg
 	SubmitSignedAggregateSelectionProof(msg *phase0.SignedAggregateAndProof) error
 }
 
 // SyncCommitteeCalls interface has all sync committee duty specific calls
 type SyncCommitteeCalls interface {
 	// GetSyncMessageBlockRoot returns beacon block root for sync committee
-	GetSyncMessageBlockRoot() (phase0.Root, error)
+	GetSyncMessageBlockRoot(slot phase0.Slot) (phase0.Root, error)
```

Comment:
not sure we need to specify, is there any situation where it's not the current slot?

Reply:
IMO it's better to use duty slot and not assume the current slot is right